lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -411,53 +398,85 @@ def calculate_wet_bulb_temperature(self, temperature, relative_humidity,
                 Cube of air pressures (Pa).
 
         Returns:
-            wbt (iris.cube.Cube):
+            iris.cube.Cube:
                 Cube of wet bulb temperature (K).
 
         """
-        precision = np.full(temperature.data.shape, self.precision)
-
-        # Set units of input diagnostics.
         relative_humidity.convert_units(1)
         pressure.convert_units('Pa')
         temperature.convert_units('K')
 
+        precision = np.full(temperature.data.shape, self.precision)
+        wbt = temperature.copy()
+        wbt.rename('wet_bulb_temperature')
+
+        latent_heat = Utilities.latent_heat_of_condensation(temperature)
+
+        wbt_data = self._calculate_wbt(
+            latent_heat.data, precision, pressure.data,
+            relative_humidity.data, temperature.data)
+
+        wbt.data = wbt_data
+        return wbt
+
+    def _calculate_wbt(self, latent_heat, precision, pressure,
```

Comment:
This says it does exactly the same as the previous function calculate_wet_bulb_temperature? Do we need both of them?